core/src/main/java/org/infinispan/persistence/sifs/IndexNode.java

```diff
@@ -375,9 +375,10 @@ private void updateFileOffsetInFile(int leafOffset, int newFile, int newOffset)
       }
       offset += (long) leafOffset * LEAF_NODE_REFERENCE_SIZE;
 
-      ByteBuffer buffer = ByteBuffer.allocate(8);
+      ByteBuffer buffer = ByteBuffer.allocate(10);
```

Comment:
This line and the one immediately after the actual fix.

The problem is when an index was updated it only updated the new file and offset, but wasn't updating the records at the same time. This would _work_ before as long as the index entry was not removed from memory or the server wasn't restarted. But however if it had to recreate the index from disk or the server was restarted it would be the incorrect number. Once this happens after compaction runs it would remove the old entries and decrement the records and think the entry is no longer in the index because the records were 0 and remove it from the index, but it was still in the data :(